lib/kubernetes-deploy/resource_watcher.rb

```diff
@@ -10,30 +10,33 @@ def initialize(resources, logger:, deploy_started_at: Time.now.utc)
       @resources = resources
       @logger = logger
       @deploy_started_at = deploy_started_at
+      @operation_name = operation_name
     end
 
-    def run(delay_sync: 3.seconds, reminder_interval: 30.seconds)
+    def run(delay_sync: 3.seconds, reminder_interval: 30.seconds, record_summary: true)
       delay_sync_until = last_message_logged_at = Time.now.utc
+      remainder = @resources.dup
```

Comment:
Was using the original object a bug?

Reply:
It would have been a bug if anything tried to reuse the same ResourceWatcher instance (but we don't). Without this change, `record_statuses_for_summary` wouldn't record anything, as `@resources` was always empty by the end of the method. Technically I don't need to `dup` here as long as the body uses a separate variable, since the variable is being replaced (not mutated) by `newly_finished_resources, remainder = remainder.partition(&:deploy_finished?)`.